base-console-app-cert/Helpers/MsalAuthenticationProvider.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Graph;
+using Microsoft.Extensions.Configuration;
+using System.Linq;
+
+namespace ConsoleGraphTest
+{
+    // This class encapsulates the details of getting a token from MSAL and exposes it via the 
+    // IAuthenticationProvider interface so that GraphServiceClient or AuthHandler can use it.
+    // A significantly enhanced version of this class will in the future be available from
+    // the GraphSDK team.  It will supports all the types of Client Application as defined by MSAL.
+    public class MsalAuthenticationProvider : IAuthenticationProvider
+    {
+        private IConfidentialClientApplication _clientApplication;
+        private string[] _scopes;
+
+        public MsalAuthenticationProvider(IConfidentialClientApplication clientApplication, string[] scopes) {
```

Comment:
This supposes that you don't change the scopes later?
this is ok for a sample, but does it scale to real life apps?

Reply:
@jmprieur By changing scopes do you mean 1) dynamic consent to additional scopes or 2) changing administratively assigned scopes on the app registration?  For the former, the majority of samples in this repo use app-only permissions in a non-interactive session thus dynamic consent doesn't apply to those scenarios.  For the latter the README files for each app detail the permissions required for each sample and are not expected to change.  If I misunderstood please let me know.

Reply:
Good point that this is a daemon. In that case you could just use `https://graph.microsoft.com/.default`